### PR TITLE
add uri-js tests

### PR DESCRIPTION
**Pull request by Uzlopak**
Current status:

> ℹ tests 32
> ℹ suites 0
> ℹ pass 21
> ℹ fail 11
> ℹ cancelled 0
> ℹ skipped 0
> ℹ todo 0
> ℹ duration_ms 29.631727

output:

```sh

aras@aras-Lenovo-Legion-5-17ARH05H:~/workspace/fast-uri$ node ./test/uri-js.test.js 
✔ Acquire URI (1.134565ms)
✔ URI Parsing (1.286119ms)
✔ URI Serialization (0.742828ms)
✖ URI Resolving (1.487959ms)
  AssertionError [ERR_ASSERTION]: /adf\ngf
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:417:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: false,
    code: 'ERR_ASSERTION',
    actual: '//www.g.com/adf\ngf',
    expected: '//www.g.com/adf%0Agf',
    operator: 'strictEqual'
  }

✖ URI Normalizing (0.892636ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://www.example.org/red%252509ros%25C3%25A9#red'
  - 'uri://www.example.org/red%09ros%C3%A9#red'
                               ^
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:423:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://www.example.org/red%252509ros%25C3%25A9#red',
    expected: 'uri://www.example.org/red%09ros%C3%A9#red',
    operator: 'strictEqual'
  }

✔ URI Equals (0.388595ms)
✖ Escape Component (0.174602ms)
  TypeError: URI.escapeComponent is not a function
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:459:23)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7)

✖ Unescape Component (0.123409ms)
  TypeError: URI.unescapeComponent is not a function
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:474:21)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7)

✖ IRI Parsing (1.314545ms)
  AssertionError [ERR_ASSERTION]: path
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:496:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: false,
    code: 'ERR_ASSERTION',
    actual: '/o%EF%A4%80ne/t%EF%B7%8Fwo.t%EF%B7%B0hree',
    expected: '/o豈ne/t﷏wo.tﷰhree',
    operator: 'strictEqual'
  }

✖ IRI Serialization (0.589806ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://us er:pa퟿ss@example.com:123/o%uF900ne/t%uFDCFwo.t%uFDF0hree?q1=a1&q2=a2#bo￯dy'
  - 'uri://us er:pa퟿ss@example.com:123/o豈ne/t﷏wo.tﷰhree?q1=a1&q2=a2#bo￯dy%EE%80%81'
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:511:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://us er:pa퟿ss@example.com:123/o%uF900ne/t%uFDCFwo.t%uFDF0hree?q1=a1&q2=a2#bo￯dy',
    expected: 'uri://us er:pa퟿ss@example.com:123/o豈ne/t﷏wo.tﷰhree?q1=a1&q2=a2#bo￯dy%EE%80%81',
    operator: 'strictEqual'
  }

✖ IRI Normalizing (0.248145ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://www.example.org/red%252509ros%25C3%25A9#red'
  - 'uri://www.example.org/red%09rosé#red'
                               ^
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:515:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://www.example.org/red%252509ros%25C3%25A9#red',
    expected: 'uri://www.example.org/red%09rosé#red',
    operator: 'strictEqual'
  }

✖ IRI Equals (0.177885ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  
  false !== true
  
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:520:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: 'strictEqual'
  }

✖ Convert IRI to URI (0.212456ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://www.example.org/red%252509ros%25C3%25A9#red'
  - 'uri://www.example.org/red%09ros%C3%A9#red'
                               ^
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:525:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://www.example.org/red%252509ros%25C3%25A9#red',
    expected: 'uri://www.example.org/red%09ros%C3%A9#red',
    operator: 'strictEqual'
  }

✖ Convert URI to IRI (0.297243ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://www.example.org/D%2525C3%2525BCrst'
  - 'uri://www.example.org/Dürst'
                            ^
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:533:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://www.example.org/D%2525C3%2525BCrst',
    expected: 'uri://www.example.org/Dürst',
    operator: 'strictEqual'
  }

✔ HTTP Equals (0.181935ms)
✔ HTTPS Equals (0.143523ms)
✔ URN Parsing (0.269656ms)
✔ URN Serialization (0.137866ms)
✖ URN Equals (0.474708ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  
  true !== false
  
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:606:5)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: true,
    expected: false,
    operator: 'strictEqual'
  }

✔ URN Resolving (0.142406ms)
✔ UUID Parsing (0.389992ms)
✔ UUID Serialization (0.137377ms)
✔ UUID Equals (0.102526ms)
✔ URN NID Override (0.112792ms)
✔ WS Parse (0.155955ms)
✔ WS Serialize (0.295636ms)
✔ WS Equal (0.120057ms)
✔ WS Normalize (0.07906ms)
✔ WSS Parse (0.158888ms)
✔ WSS Serialize (0.137447ms)
✔ WSS Equal (0.085764ms)
✔ WSS Normalize (0.070539ms)
ℹ tests 32
ℹ suites 0
ℹ pass 21
ℹ fail 11
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 29.631727

✖ failing tests:

test at test/uri-js.test.js:362:1
✖ URI Resolving (1.487959ms)
  AssertionError [ERR_ASSERTION]: /adf\ngf
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:417:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: false,
    code: 'ERR_ASSERTION',
    actual: '//www.g.com/adf\ngf',
    expected: '//www.g.com/adf%0Agf',
    operator: 'strictEqual'
  }

test at test/uri-js.test.js:421:1
✖ URI Normalizing (0.892636ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://www.example.org/red%252509ros%25C3%25A9#red'
  - 'uri://www.example.org/red%09ros%C3%A9#red'
                               ^
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:423:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://www.example.org/red%252509ros%25C3%25A9#red',
    expected: 'uri://www.example.org/red%09ros%C3%A9#red',
    operator: 'strictEqual'
  }

test at test/uri-js.test.js:454:1
✖ Escape Component (0.174602ms)
  TypeError: URI.escapeComponent is not a function
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:459:23)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7)

test at test/uri-js.test.js:470:1
✖ Unescape Component (0.123409ms)
  TypeError: URI.unescapeComponent is not a function
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:474:21)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7)

test at test/uri-js.test.js:488:1
✖ IRI Parsing (1.314545ms)
  AssertionError [ERR_ASSERTION]: path
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:496:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: false,
    code: 'ERR_ASSERTION',
    actual: '/o%EF%A4%80ne/t%EF%B7%8Fwo.t%EF%B7%B0hree',
    expected: '/o豈ne/t﷏wo.tﷰhree',
    operator: 'strictEqual'
  }

test at test/uri-js.test.js:501:1
✖ IRI Serialization (0.589806ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://us er:pa퟿ss@example.com:123/o%uF900ne/t%uFDCFwo.t%uFDF0hree?q1=a1&q2=a2#bo￯dy'
  - 'uri://us er:pa퟿ss@example.com:123/o豈ne/t﷏wo.tﷰhree?q1=a1&q2=a2#bo￯dy%EE%80%81'
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:511:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://us er:pa퟿ss@example.com:123/o%uF900ne/t%uFDCFwo.t%uFDF0hree?q1=a1&q2=a2#bo￯dy',
    expected: 'uri://us er:pa퟿ss@example.com:123/o豈ne/t﷏wo.tﷰhree?q1=a1&q2=a2#bo￯dy%EE%80%81',
    operator: 'strictEqual'
  }

test at test/uri-js.test.js:514:1
✖ IRI Normalizing (0.248145ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://www.example.org/red%252509ros%25C3%25A9#red'
  - 'uri://www.example.org/red%09rosé#red'
                               ^
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:515:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://www.example.org/red%252509ros%25C3%25A9#red',
    expected: 'uri://www.example.org/red%09rosé#red',
    operator: 'strictEqual'
  }

test at test/uri-js.test.js:518:1
✖ IRI Equals (0.177885ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  
  false !== true
  
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:520:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: 'strictEqual'
  }

test at test/uri-js.test.js:523:1
✖ Convert IRI to URI (0.212456ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://www.example.org/red%252509ros%25C3%25A9#red'
  - 'uri://www.example.org/red%09ros%C3%A9#red'
                               ^
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:525:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://www.example.org/red%252509ros%25C3%25A9#red',
    expected: 'uri://www.example.org/red%09ros%C3%A9#red',
    operator: 'strictEqual'
  }

test at test/uri-js.test.js:531:1
✖ Convert URI to IRI (0.297243ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  + actual - expected
  
  + 'uri://www.example.org/D%2525C3%2525BCrst'
  - 'uri://www.example.org/Dürst'
                            ^
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:533:3)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: 'uri://www.example.org/D%2525C3%2525BCrst',
    expected: 'uri://www.example.org/Dürst',
    operator: 'strictEqual'
  }

test at test/uri-js.test.js:601:3
✖ URN Equals (0.474708ms)
  AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
  
  true !== false
  
      at TestContext.<anonymous> (/home/aras/workspace/fast-uri/test/uri-js.test.js:606:5)
      at Test.runInAsyncScope (node:async_hooks:206:9)
      at Test.run (node:internal/test_runner/test:749:25)
      at Test.processPendingSubtests (node:internal/test_runner/test:463:18)
      at Test.postRun (node:internal/test_runner/test:848:19)
      at Test.run (node:internal/test_runner/test:791:12)
      at async Test.processPendingSubtests (node:internal/test_runner/test:463:7) {
    generatedMessage: true,
    code: 'ERR_ASSERTION',
    actual: true,
    expected: false,
    operator: 'strictEqual'
  }
```

### CODE REVIEW

File: test/uri-js.test.js [hunk 1]
```diff
@@ -0,0 +1,928 @@
...
+ *
+ * The views and conclusions contained in the software and documentation are those of the
+ * authors and should not be interpreted as representing official policies, either expressed
+ * or implied, of Gary Court.
+ */
+
+test('Acquire URI', function () {
+  // URI = require("./uri").URI;
+  ok(URI)
+})
```
Comment (mcollina) [inline test/uri-js.test.js:54]: This seems useless. can you remove?

**Review by mcollina — APPROVED** [review]
+1

Could you skip the one that are failing so we can land them?

Or fix them all.

File: test/uri-js.test.js (hunk 1 above)
Comment (Uzlopak) [reply test/uri-js.test.js:54]: Its basically the original uri-js test file copy pasted into that file. Everything you see after the license is from the original test file. Just reformatted and fixed the 10 linting issues.

**Comment by gurgunday** [comment]
I will start working on them right after the merge

**Comment by Uzlopak** [comment]
@mcollina 
@gurgunday 

skipping failing tests

Also see for detected issues #90 and #89

**Comment by mcollina** [comment]
CI is failing on Node.js v14

**Comment by Uzlopak** [comment]
@mcollina 

rewrote the test harness to use tap

**Review by gurgunday — APPROVED** [review]

**Pull request merged**